### PR TITLE
IBP-2770-UseFieldParentValueAsFieldMapLocation

### PR DESCRIPTION
**Pull request by gelli-leafnode**
Remove override process for location if location is not null || eq. to 1
Modified query in getAllFieldMapsInBlockByTrialInstanceId - return fieldmap parent as sitelocationid 
Fix failing unit test

### CODE REVIEW

File: src/main/java/org/generationcp/middleware/dao/dms/ExperimentPropertyDao.java [hunk 1]
```diff
@@ -391,8 +391,8 @@
 				trial.setInstanceId((Integer) row[3]);
 				trial.setSiteName((String) row[4]);
 				trial.setLocationName((String) row[4]);
-				if (row[5] != null && NumberUtils.isNumber((String) row[5])) {
-					trial.setLocationId(Integer.valueOf((String) row[5]));
+				if (row[5] != null && NumberUtils.isNumber(row[5].toString())) {
```
Comment (abatac) [inline src/main/java/org/generationcp/middleware/dao/dms/ExperimentPropertyDao.java:394]: Hi @gelli-leafnode , Use String.valueOf to prevent NPE in toString() when object is null

File: src/main/java/org/generationcp/middleware/operation/saver/GeolocationPropertySaver.java [hunk 2]
```diff
@@ -26,9 +26,6 @@ public void saveFieldmapProperties(final List<FieldMapInfo> infos) throws Middle
 					int locationId = trial.getInstanceId();
 					if (trial.getInstanceId() != null && trial.getInstanceId().intValue() == 1) {
 						locationId = this.getExperimentModelSaver().moveStudyToNewGeolocation(info.getFieldbookId());
-					}
```
Comment (abatac) [inline src/main/java/org/generationcp/middleware/operation/saver/GeolocationPropertySaver.java:29]: Why did you remove null checking for trial getLocationId?

**Review by abatac — CHANGES_REQUESTED** [review]

File: src/main/java/org/generationcp/middleware/operation/saver/GeolocationPropertySaver.java (hunk 2 above)
Comment (gelli-leafnode) [reply src/main/java/org/generationcp/middleware/operation/saver/GeolocationPropertySaver.java:29]: Hi @abatac, will revert checking if location is not null before saving

**Review by abatac — APPROVED** [review]
